encoding/ccf/encode.go

```diff
@@ -943,6 +1037,18 @@ func (e *Encoder) encodeComposite(
 		return err
 	}
 
+	if e.em.sortCompositeFields == SortNone {
```

Comment:
Here, and for the similar code below: `EnforceSortMode` is an enum. It currently only has two cases, so checking for the first and returning, then implicitly continuing for the second case works. Maybe make the handling of each case explicit by using a switch case, and panic with `NewUnreachableError` in the default case.